### PR TITLE
Prepare for swmr and mpiio

### PR DESCRIPTION
**Pull request by dominikwerder**
As preparation for SWMR and parallel writing, the HDF file and the HDF objects are created, then the file is closed and re-opened as required by the HDF API. The classes `CommandHandler` and `HDFFile` therefore had to change.
A new helper `Msg` is added in order to unify message handling and to allow better performance testing.
New types, which will only be used in SWMR and parallel, include `HDFIDStore` and `CollectiveQueue` and are added here already because they appear in the public interface of `HDFWriterModule`.

### CODE REVIEW

File: src/tests/HDFFile.cxx [hunk 1]
```diff
@@ -516,17 +527,39 @@ class T_CommandHandler : public testing::Test {
...
+        "fname": "tmp-mmap"
+      },
+      "mpi": {
+        "path_bin": "."
+      },
+      "shm": 2100100100
     })"");
 
+    // TODO
+    // This must go somewhere else...
```
Comment (matthew-d-jones) [inline src/tests/HDFFile.cxx]: Github issue

Comment (matthew-d-jones) [reply src/tests/HDFFile.cxx]: Created issue #106

File: src/schemas/f142/f142_rw.cxx [hunk 2]
```diff
@@ -185,155 +241,363 @@ FileWriter::HDFWriterModule::ptr HDFWriterModule::create() {
...
+  }
+  if (type == "double") {
+    return Value::ArrayDouble;
+  }
+  return Value::Int;
+}
+
+// clang-format: off
+
+// should make this templated..
```
[the remark this answers is not in the document]
Comment (matthew-d-jones) [reply src/schemas/f142/f142_rw.cxx]: Created issue #106

File: src/h5.cxx [hunk 3]
```diff
@@ -113,35 +114,148 @@ h5s::~h5s() {
...
+  if (dsp_mem != -1) {
+    H5Sclose(dsp_mem);
+    dsp_mem = -1;
+  }
+  if (pl_transfer != -1) {
+    H5Pclose(pl_transfer);
+    pl_transfer = -1;
+  }
+  if (type != -1) {
+    // TODO
```
[the remark this answers is not in the document]
Comment (matthew-d-jones) [reply src/h5.cxx]: Created issue #106

**Comment by martukas** [comment]
I looked at @matthew-d-jones changes. Things look fine for now. I'm happy to take it over in this state. We can refactor other aspects of it later.

**Pull request merged**